File: frontend/src/lib/api/notifications.ts
import { api } from "./client";

export interface InAppNotification {
  id: string;
  kind: string;
  subject_type: string;
  subject_id: string;
  payload: Record<string, unknown>;
  read_at: string | null;
  created_at: string;
}

export interface NotificationsList {
  items: InAppNotification[];
  unread_count: number;
}

export const listNotifications = (unread = false) =>
  api<NotificationsList>(`/api/me/notifications${unread ? "?unread=true" : ""}`);

export const markRead = (id: string) =>
  api<void>(`/api/me/notifications/${id}/read`, { method: "POST" });

export const readAll = () =>
  api<void>("/api/me/notifications/read-all", { method: "POST" });

File: packages/core/src/setvault_core/schemas/notifications.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class InAppNotificationOut(BaseModel):
    id: str
    kind: str
    subject_type: str
    subject_id: str
    payload: dict
    read_at: datetime | None = None
    created_at: datetime


class NotificationsListOut(BaseModel):
    items: list[InAppNotificationOut]
    unread_count: int
